frontend/src/lib/auth/index.ts

```diff
@@ -86,3 +92,95 @@ export async function getRequestToken() {
     secret: params.get("oauth_token_secret"),
   };
 }
+
+export const auth = async ({
+  token,
+  secret,
+}: {
+  token?: string | null;
+  secret?: string | null;
+}) => {
+  const cookies = await cookiesPromise();
+  const accessToken = token ?? cookies.get("access_token")?.value;
+  const accessSecret = secret ?? cookies.get("access_token_secret")?.value;
+
+  if (accessToken === "" || accessSecret === "") {
+    return false;
+  }
+
+  try {
+    const response = await fetch(`${env.NEXT_PUBLIC_API_URL}/user/login`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ accessToken, accessSecret }),
+      credentials: "include",
+    });
+    const data = (await response.json()) as { error?: string };
+    if (data.error !== undefined) {
+      cookies.delete({
+        name: "access_token",
+        path: "/",
+      });
+      cookies.delete({
+        name: "access_token_secret",
+        path: "/",
+      });
+      throw new Error(data.error);
+    }
+    const setCookieHeaders = response.headers.getSetCookie();
+    setCookieHeaders.forEach((setCookie) => {
+      const name = setCookie.split("=")[0];
+      const value = setCookie.split("=")[1].split(";")[0];
```

Comment:
```suggestion
      const [name, value] = setCookie.split("=");
```

i wytłumacz w komentarzu o co chodzi z tym splitem na `;` bo nie rozumiem

Reply:
bo masz ze "key=value;" wiec ucinam ten ;, rownie dobrze moge slicem